rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/RemoteFetchProcessor.java

```diff
@@ -94,20 +108,49 @@ public void addToProcessQueue(String remotePath, long offset, int length, long f
     totalDownloadRequests.inc();
   }
 
+  public void addToProcessQueueSafe(String remotePath, Set<Range<Long>> closedOpenRanges, long fileSize, long lastModified)
+  {
+    try {
+      addToProcessQueue(remotePath, closedOpenRanges, fileSize, lastModified);
+    }
+    catch (Exception e) {
+      log.warn("Unable to queue ranges for file: " + remotePath);
+    }
+  }
+
+  public void addToProcessQueue(String remotePath, Set<Range<Long>> closedOpenRanges, long fileSize, long lastModified)
```

Comment:
nit: this can be declared as a private method.

Reply:
will change this.